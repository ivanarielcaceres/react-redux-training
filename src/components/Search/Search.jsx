import React from 'react';
import PropTypes from 'prop-types';
import { FaSearch } from 'react-icons/fa';
import { Button, Input, Row, Col } from 'reactstrap';

export default class Search extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      keyword: ''
    };
  }

  search = () => {
    console.log('keyword', this.state.keyword);
    this.props.doSearch(this.state.keyword);
  };

  render = () => {
    return (
      <div>
        <Row>
          <Col>
            <Input
              placeholder="keyword"
              value={this.state.keyword}
              onChange={e => this.setState({ keyword: e.target.value })}
            />
          </Col>
          <Col>
            <Button onClick={this.search}>
              {' '}
              <FaSearch />{' '}
            </Button>
          </Col>
        </Row>
      </div>
    );
  };
}
