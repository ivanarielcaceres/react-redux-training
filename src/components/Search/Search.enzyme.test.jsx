import React from 'react';
import { shallow } from 'enzyme';
import Search from 'components/Search/Search';
import { Button, Input } from 'reactstrap';

/**
 * Smoke test
 * */
describe('<Search />', () => {
  let wrapper;
  const mockLoginfn = jest.fn();
  beforeEach(() => {
    wrapper = shallow(<Search doSearch={mockLoginfn} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('Render without crasing', () => {
    shallow(<Search doSearch={mockLoginfn} />);
  });

  it('Simulate Search', () => {
    const search = () => {
      wrapper
        .find(Button)
        .at(0)
        .simulate('click');
    };
    //Simulate Input value change
    wrapper
      .find(Input)
      .at(0)
      .simulate('change', { target: { value: 'Item 0' } });
    //Searching using Input value
    search();
  });
});
