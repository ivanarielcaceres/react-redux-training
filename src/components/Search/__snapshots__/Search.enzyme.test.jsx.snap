// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> Render without crasing 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    doSearch={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Row
        tag="div"
>
        <Col
                tag="div"
                widths={
                        Array [
                                "xs",
                                "sm",
                                "md",
                                "lg",
                                "xl",
                              ]
                }
        >
                <Input
                        onChange={[Function]}
                        placeholder="keyword"
                        type="text"
                        value=""
                />
        </Col>
        <Col
                tag="div"
                widths={
                        Array [
                                "xs",
                                "sm",
                                "md",
                                "lg",
                                "xl",
                              ]
                }
        >
                <Button
                        color="secondary"
                        onClick={[Function]}
                        tag="button"
                >
                         
                        <Unknown />
                         
                </Button>
        </Col>
</Row>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Col
            tag="div"
            widths={
                        Array [
                                    "xs",
                                    "sm",
                                    "md",
                                    "lg",
                                    "xl",
                                  ]
            }
>
            <Input
                        onChange={[Function]}
                        placeholder="keyword"
                        type="text"
                        value=""
            />
</Col>,
          <Col
            tag="div"
            widths={
                        Array [
                                    "xs",
                                    "sm",
                                    "md",
                                    "lg",
                                    "xl",
                                  ]
            }
>
            <Button
                        color="secondary"
                        onClick={[Function]}
                        tag="button"
            >
                         
                        <Unknown />
                         
            </Button>
</Col>,
        ],
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Input
              onChange={[Function]}
              placeholder="keyword"
              type="text"
              value=""
/>,
            "tag": "div",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onChange": [Function],
              "placeholder": "keyword",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Button
              color="secondary"
              onClick={[Function]}
              tag="button"
>
               
              <Unknown />
               
</Button>,
            "tag": "div",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                " ",
                <Unknown />,
                " ",
              ],
              "color": "secondary",
              "onClick": [Function],
              "tag": "button",
            },
            "ref": null,
            "rendered": Array [
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              " ",
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Row
          tag="div"
>
          <Col
                    tag="div"
                    widths={
                              Array [
                                        "xs",
                                        "sm",
                                        "md",
                                        "lg",
                                        "xl",
                                      ]
                    }
          >
                    <Input
                              onChange={[Function]}
                              placeholder="keyword"
                              type="text"
                              value=""
                    />
          </Col>
          <Col
                    tag="div"
                    widths={
                              Array [
                                        "xs",
                                        "sm",
                                        "md",
                                        "lg",
                                        "xl",
                                      ]
                    }
          >
                    <Button
                              color="secondary"
                              onClick={[Function]}
                              tag="button"
                    >
                               
                              <Unknown />
                               
                    </Button>
          </Col>
</Row>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Col
              tag="div"
              widths={
                            Array [
                                          "xs",
                                          "sm",
                                          "md",
                                          "lg",
                                          "xl",
                                        ]
              }
>
              <Input
                            onChange={[Function]}
                            placeholder="keyword"
                            type="text"
                            value=""
              />
</Col>,
            <Col
              tag="div"
              widths={
                            Array [
                                          "xs",
                                          "sm",
                                          "md",
                                          "lg",
                                          "xl",
                                        ]
              }
>
              <Button
                            color="secondary"
                            onClick={[Function]}
                            tag="button"
              >
                             
                            <Unknown />
                             
              </Button>
</Col>,
          ],
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Input
                onChange={[Function]}
                placeholder="keyword"
                type="text"
                value=""
/>,
              "tag": "div",
              "widths": Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onChange": [Function],
                "placeholder": "keyword",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Button
                color="secondary"
                onClick={[Function]}
                tag="button"
>
                 
                <Unknown />
                 
</Button>,
              "tag": "div",
              "widths": Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  " ",
                  <Unknown />,
                  " ",
                ],
                "color": "secondary",
                "onClick": [Function],
                "tag": "button",
              },
              "ref": null,
              "rendered": Array [
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                " ",
              ],
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<Search /> Simulate Search 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    doSearch={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Row
        tag="div"
>
        <Col
                tag="div"
                widths={
                        Array [
                                "xs",
                                "sm",
                                "md",
                                "lg",
                                "xl",
                              ]
                }
        >
                <Input
                        onChange={[Function]}
                        placeholder="keyword"
                        type="text"
                        value=""
                />
        </Col>
        <Col
                tag="div"
                widths={
                        Array [
                                "xs",
                                "sm",
                                "md",
                                "lg",
                                "xl",
                              ]
                }
        >
                <Button
                        color="secondary"
                        onClick={[Function]}
                        tag="button"
                >
                         
                        <Unknown />
                         
                </Button>
        </Col>
</Row>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Col
            tag="div"
            widths={
                        Array [
                                    "xs",
                                    "sm",
                                    "md",
                                    "lg",
                                    "xl",
                                  ]
            }
>
            <Input
                        onChange={[Function]}
                        placeholder="keyword"
                        type="text"
                        value=""
            />
</Col>,
          <Col
            tag="div"
            widths={
                        Array [
                                    "xs",
                                    "sm",
                                    "md",
                                    "lg",
                                    "xl",
                                  ]
            }
>
            <Button
                        color="secondary"
                        onClick={[Function]}
                        tag="button"
            >
                         
                        <Unknown />
                         
            </Button>
</Col>,
        ],
        "tag": "div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Input
              onChange={[Function]}
              placeholder="keyword"
              type="text"
              value=""
/>,
            "tag": "div",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "onChange": [Function],
              "placeholder": "keyword",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <Button
              color="secondary"
              onClick={[Function]}
              tag="button"
>
               
              <Unknown />
               
</Button>,
            "tag": "div",
            "widths": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                " ",
                <Unknown />,
                " ",
              ],
              "color": "secondary",
              "onClick": [Function],
              "tag": "button",
            },
            "ref": null,
            "rendered": Array [
              " ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              " ",
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Row
          tag="div"
>
          <Col
                    tag="div"
                    widths={
                              Array [
                                        "xs",
                                        "sm",
                                        "md",
                                        "lg",
                                        "xl",
                                      ]
                    }
          >
                    <Input
                              onChange={[Function]}
                              placeholder="keyword"
                              type="text"
                              value=""
                    />
          </Col>
          <Col
                    tag="div"
                    widths={
                              Array [
                                        "xs",
                                        "sm",
                                        "md",
                                        "lg",
                                        "xl",
                                      ]
                    }
          >
                    <Button
                              color="secondary"
                              onClick={[Function]}
                              tag="button"
                    >
                               
                              <Unknown />
                               
                    </Button>
          </Col>
</Row>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Col
              tag="div"
              widths={
                            Array [
                                          "xs",
                                          "sm",
                                          "md",
                                          "lg",
                                          "xl",
                                        ]
              }
>
              <Input
                            onChange={[Function]}
                            placeholder="keyword"
                            type="text"
                            value=""
              />
</Col>,
            <Col
              tag="div"
              widths={
                            Array [
                                          "xs",
                                          "sm",
                                          "md",
                                          "lg",
                                          "xl",
                                        ]
              }
>
              <Button
                            color="secondary"
                            onClick={[Function]}
                            tag="button"
              >
                             
                            <Unknown />
                             
              </Button>
</Col>,
          ],
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Input
                onChange={[Function]}
                placeholder="keyword"
                type="text"
                value=""
/>,
              "tag": "div",
              "widths": Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "onChange": [Function],
                "placeholder": "keyword",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Button
                color="secondary"
                onClick={[Function]}
                tag="button"
>
                 
                <Unknown />
                 
</Button>,
              "tag": "div",
              "widths": Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  " ",
                  <Unknown />,
                  " ",
                ],
                "color": "secondary",
                "onClick": [Function],
                "tag": "button",
              },
              "ref": null,
              "rendered": Array [
                " ",
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                " ",
              ],
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
